transcom/mymove
# B 20757 INT - V2

## [B-20757](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A983881)

## Summary

This task updates the error message for DTOD so it is clear why the PPM cannot progress. Follow-on tasks will provide a more granular error message based on reason for failure.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

NOTE: Prior to test I recommend adding "distanceFloat = -1" after line 121 of dtod_zip5_distance.go to simulate a DTOD failure, or use the zip 23471 as either pickup or destination. It is a known bad zip.

1. Access the milmove application
2. Login as a Customer
3. Create a PPM
4. Fill in all field for the ppm and progress to the estimated incentives page (use bad zips if you have them, refer to testing note for guidance on how to simulate failures)
5. Fill in estimated weight
6. Save and Continue
7. Verify that error message matches the error message in the acceptance criteria for [B-20757](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A983881)
"We are unable to calculate your distance. It may be that you have entered an invalid ZIP Code, or the system that calculates distance (DTOD) may be down. Please check your ZIP Code to ensure it was entered correctly and is not a PO Box. If the error persists, please try again later, or contact the Technical Help Desk."

1. Access the office application
2. Login as a Service Counselor
3. Select a Move
4. Create a PPM
5. Fill in all field for the ppm and progress to the estimated incentives page (use bad zips like 23471, refer to testing note for guidance on how to simulate failures)
7. Save and Continue
9. Verify that error message matches the error message in the acceptance criteria for [B-20757](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A983881)
"We are unable to calculate your distance. It may be that you have entered an invalid ZIP Code, or the system that calculates distance (DTOD) may be down. Please check your ZIP Code to ensure it was entered correctly and is not a PO Box. If the error persists, please try again later, or contact the Technical Help Desk."

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
